exporters/etw/include/opentelemetry/exporters/etw/etw_provider_exporter.h

```diff
@@ -0,0 +1,664 @@
+        break;
+      case common::AttributeType::TYPE_CSTRING:
+        eventName = (char *)(nostd::get<const char *>(nameField));
+        break;
+      default:
+        // This is user error. Invalid event name!
+        // We supply default 'NoName' event name in this case.
+        break;
+    }
+
+    /* clang-format off */
+    nlohmann::json jObj =
+    {
+      { "env_name", "Span" },
+      { "env_ver", "4.0" },
+
+      // TODO: remove these fields. Provided for illustrative purposes only.
+      { "env_cloud_role", "BusyWorker" },
+      { "env_cloud_roleInstance", "CY1SCH030021417" },
+      { "env_cloud_roleVer", "9.0.15289.2" },
+      //
+
+      // TODO: compute time in MessagePack-friendly format
+      // TODO: should we consider uint64_t format with Unix timestamps for ELK stack?
+      { "env_time",
+        {
+          { "TypeCode", 255 },
+          { "Body","0xFFFFFC60000000005F752C2C" }
+        }
+      },
+      //
+
+      // TODO: follow JSON implementation of OTLP or place protobuf for non-Microsoft flows
+      { "env_dt_traceId", "6dcdae7b9b0c7643967d74ee54056178" },
+      { "env_dt_spanId", "5866c4322919e641" },
+      //
+
+      { "name", eventName },
+      { "kind", 0 },
+      { "startTime",
+        {
+          // TODO: timestamp
+          { "TypeCode", 255 },
+          { "Body", "0xFFFF87CC000000005F752C2C" }
+        }
+      }
+    };
+    /* clang-format on */
+
+    for (auto &kv : eventData)
+    {
+      const char *name = kv.first.data();
+      // Don't include event name field in the payload
+      if (EVENT_NAME == name)
+        continue;
+      auto &value = kv.second;
+      switch (value.index())
+      {
+        case common::AttributeType::TYPE_BOOL:
+        {
+          UINT8 temp = static_cast<UINT8>(nostd::get<bool>(value));
+          jObj[name] = temp;
+          break;
+        }
+        case common::AttributeType::TYPE_INT:
+        {
+          auto temp  = nostd::get<int32_t>(value);
+          jObj[name] = temp;
+          break;
+        }
+        case common::AttributeType::TYPE_INT64:
+        {
+          auto temp  = nostd::get<int64_t>(value);
+          jObj[name] = temp;
+          break;
+        }
+        case common::AttributeType::TYPE_UINT:
+        {
+          auto temp  = nostd::get<uint32_t>(value);
+          jObj[name] = temp;
+          break;
+        }
+        case common::AttributeType::TYPE_UINT64:
+        {
+          auto temp  = nostd::get<uint64_t>(value);
+          jObj[name] = temp;
+          break;
+        }
+        case common::AttributeType::TYPE_DOUBLE:
+        {
+          auto temp  = nostd::get<double>(value);
+          jObj[name] = temp;
+          break;
+        }
+        case common::AttributeType::TYPE_STRING:
+        {
+          auto temp  = nostd::get<nostd::string_view>(value);
+          jObj[name] = temp;
+          break;
+        }
+        case common::AttributeType::TYPE_CSTRING:
```

Comment:
Watch out for this one - you may need to comment that line, OR wait for my other PR. Currently the `CSTRING` (`const char *`) is NOT exactly supported by other exporters.